### PR TITLE
[Merged by Bors] - chore: update 'section binary' in Init/Logic to use infix notation and `calc`

### PR DESCRIPTION
**Pull request by dwrensha**
This makes the code more closely match the mathlib3 version: https://github.com/leanprover-community/lean/blob/741670c439f1ca266bc7fe61ef7212cc9afd9dd8/library/init/logic.lean#L1064-L1097

### CODE REVIEW

File: scripts/nolints.json [hunk 1]
```diff
@@ -72,7 +72,6 @@
  ["docBlame", "tactic_"],
  ["docBlame", "«termListΠ»"],
  ["docBlame", "«termListΣ»"],
- ["docBlame", "«term_⁻¹»"],
```
Comment (dwrensha) [inline scripts/nolints.json:75]: Adding the new local notation meant that I would have needed to renamed this to `«term_⁻¹_1»` to get it to still silence the linter. I opted instead to just write a doc comment at the definition of the syntax.

File: Mathlib/Algebra/Group/Defs.lean [hunk 2]
```diff
@@ -39,6 +39,7 @@ instance One.ofOfNat1 {α} [OfNat α (nat_lit 1)] : One α where
 class Inv (α : Type u) where
   inv : α → α
 
+/-- `x⁻¹` is the multiplicative inverse of `x`. -/
```
Comment (dwrensha) [inline Mathlib/Algebra/Group/Defs.lean:42]: See my comment about the change in nolints.json: https://github.com/leanprover-community/mathlib4/pull/386/files#r954269146

Comment (digama0) [reply Mathlib/Algebra/Group/Defs.lean:42]: Not sure if `inheritDoc` is available on the current version of lean used by mathlib4, but if it is, the preferred solution here is to mark the syntax as `@[inheritDoc]` and put the doc comment on `Inv.inv` instead.

Comment (dwrensha) [reply Mathlib/Algebra/Group/Defs.lean]: Done. I think https://github.com/leanprover-community/mathlib4/pull/374 made `inheritDoc` available.

Comment (dwrensha) [reply Mathlib/Algebra/Group/Defs.lean]: I made the documentation a bit more generic, because I'm not sure what happens with `to_additive`.

**Comment by gebner** [comment]
Thanks!

bors r+

**Pull request closed without merging**